File: src/windows/AmbientContextMcp.Core.Tests/TransmissionUiCatalogTests.cs
using AmbientContextMcp.Core.Models;
using Xunit;

namespace AmbientContextMcp.Core.Tests;

public class TransmissionUiCatalogTests
{
    [Fact]
    public void Every_ui_option_links_only_classified_paths()
    {
        var classifications = AmbientContextCatalog.GetPrivacyClassifications()
            .Select(item => item.Path)
            .ToHashSet(StringComparer.OrdinalIgnoreCase);

        foreach (var group in AmbientContextCatalog.GetTransmissionUiGroups())
        {
            foreach (var option in group.Options)
            {
                Assert.NotEmpty(option.LinkedPaths);
                foreach (var path in option.LinkedPaths)
                {
                    Assert.Contains(path, classifications);
                }
            }
        }
    }

    [Fact]
    public void Flattened_transmission_options_cover_all_ui_linked_paths()
    {
        var uiPaths = AmbientContextCatalog.GetTransmissionUiGroups()
            .SelectMany(group => group.Options)
            .SelectMany(option => option.LinkedPaths)
            .ToHashSet(StringComparer.OrdinalIgnoreCase);
        var flatPaths = AmbientContextCatalog.GetTransmissionOptions()
            .Select(option => option.Path)
            .ToHashSet(StringComparer.OrdinalIgnoreCase);

        Assert.Equal(uiPaths, flatPaths);
    }

    [Fact]
    public void Ui_groups_have_expected_structure()
    {
        var groups = AmbientContextCatalog.GetTransmissionUiGroups();

        Assert.Equal(4, groups.Count);
        Assert.Equal(["foregroundApp", "activity", "media", "environment"], groups.Select(group => group.Id));
        Assert.Equal(11, groups.SelectMany(group => group.Options).Count());
        Assert.All(
            groups.SelectMany(group => group.Options),
            option => Assert.False(string.IsNullOrWhiteSpace(option.PrimaryPath)));
    }
}

File: src/macos/Tests/AmbientContextCoreTests/TransmissionUiCatalogTests.swift
import Testing
@testable import AmbientContextCore

@Suite("TransmissionUiCatalog")
struct TransmissionUiCatalogTests {
    @Test("Every_ui_option_links_only_classified_paths")
    func everyUiOptionLinksOnlyClassifiedPaths() {
        let classifications = Set(
            AmbientContextCatalog.getPrivacyClassifications(language: "en")
                .map { $0.path.lowercased() })

        for group in AmbientContextCatalog.getTransmissionUiGroups(language: "en") {
            for option in group.options {
                #expect(!option.linkedPaths.isEmpty)
                for path in option.linkedPaths {
                    #expect(classifications.contains(path.lowercased()))
                }
            }
        }
    }

    @Test("Flattened_transmission_options_cover_all_ui_linked_paths")
    func flattenedTransmissionOptionsCoverAllUiLinkedPaths() {
        let uiPaths = Set(
            AmbientContextCatalog.getTransmissionUiGroups(language: "en")
                .flatMap { $0.options }
                .flatMap { $0.linkedPaths }
                .map { $0.lowercased() })
        let flatPaths = Set(
            AmbientContextCatalog.getTransmissionOptions(language: "en")
                .map { $0.path.lowercased() })

        #expect(uiPaths == flatPaths)
    }

    @Test("Ui_groups_have_expected_structure")
    func uiGroupsHaveExpectedStructure() {
        let groups = AmbientContextCatalog.getTransmissionUiGroups(language: "en")

        #expect(groups.count == 4)
        #expect(groups.map(\.id) == ["foregroundApp", "activity", "media", "environment"])
        #expect(groups.flatMap(\.options).count == 11)
        for option in groups.flatMap(\.options) {
            #expect(!option.primaryPath.trimmingCharacters(in: .whitespaces).isEmpty)
        }
    }
}
